### DISCUSSION TITLE
Fleet is missing from my SOC

### DISCUSSION THREAD

**Discussion by Rbartram** [Security-Onion-Solutions/securityonion#5941] [Unsupported Versions] [answered]
Hello, I'm about to embark on deploying osquery for File Integrity Monitoring (FIM) and was looking for Fleet.  I'm pretty sure that I used to see it under Tools in the SOC.

- so-fleet was not showing on the manager when I ran so-status 
- I ran so-docker-refresh which returned the following related to so-fleet:

	Downloading so-fleet
	Executing command with retry support: docker pull ghcr.io/security-onion-solutions/so-fleet:2.3.70
	Results: 2.3.70: Pulling from security-onion-solutions/so-fleet
	188c0c94c7c5: Already exists
	41c5b03fd78b: Pulling fs layer
	8fa2c1921f17: Pulling fs layer
	cca299067502: Pulling fs layer
	b713cfff8c30: Pulling fs layer
	b713cfff8c30: Waiting
	cca299067502: Verifying Checksum
	cca299067502: Download complete
	41c5b03fd78b: Verifying Checksum
	41c5b03fd78b: Download complete
	b713cfff8c30: Verifying Checksum
	b713cfff8c30: Download complete
	41c5b03fd78b: Pull complete
	8fa2c1921f17: Verifying Checksum
	8fa2c1921f17: Download complete
	8fa2c1921f17: Pull complete
	cca299067502: Pull complete
	b713cfff8c30: Pull complete
	Digest: sha256:971bc1f8644365a98dd5b47b60b32ed12f0c98048233ecf00bd770a386185f38
	Status: Downloaded newer image for ghcr.io/security-onion-solutions/so-fleet:2.3.70
	ghcr.io/security-onion-solutions/so-fleet:2.3.70 (0)
	  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
									 Dload  Upload   Total   Spent    Left  Speed
	100   543  100   543    0     0   1792      0 --:--:-- --:--:-- --:--:--  1798
	The push refers to repository [igt-pgh-som-01:5000/security-onion-solutions/so-fleet]
	0bedc07d873f: Layer already exists
	eb0a39b57e41: Layer already exists
	5493c054586b: Layer already exists
	b0d2d72c616d: Layer already exists
	ace0eda3e3be: Layer already exists
	2.3.70: digest: sha256:d898adc9816b244e36f66ebdd44516e7ea8770a8eb7af5ae9f459ff92ff68f11 size: 1365
	Downloading so-fleet-launcher
	Executing command with retry support: docker pull ghcr.io/security-onion-solutions/so-fleet-launcher:2.3.70
	Results: 2.3.70: Pulling from security-onion-solutions/so-fleet-launcher
	e7ae86ffe2df: Already exists
	a30282bfab9d: Pulling fs layer
	ead50cc843d5: Pulling fs layer
	48f3267b40d7: Pulling fs layer
	451400f89bf3: Pulling fs layer
	fc22fa030874: Pulling fs layer
	1609b09d257f: Pulling fs layer
	fc22fa030874: Waiting
	1609b09d257f: Waiting
	451400f89bf3: Waiting
	48f3267b40d7: Verifying Checksum
	48f3267b40d7: Download complete
	ead50cc843d5: Verifying Checksum
	ead50cc843d5: Download complete
	fc22fa030874: Verifying Checksum
	fc22fa030874: Download complete
	1609b09d257f: Verifying Checksum
	1609b09d257f: Download complete
	a30282bfab9d: Verifying Checksum
	a30282bfab9d: Download complete
	451400f89bf3: Verifying Checksum
	451400f89bf3: Download complete
	a30282bfab9d: Pull complete
	ead50cc843d5: Pull complete
	48f3267b40d7: Pull complete
	451400f89bf3: Pull complete
	fc22fa030874: Pull complete
	1609b09d257f: Pull complete
	Digest: sha256:7f039aa32cb2ce9998eacc83d6d196cccc26ebbdbd3f695da8f22b0e9da8403b
	Status: Downloaded newer image for ghcr.io/security-onion-solutions/so-fleet-launcher:2.3.70
	ghcr.io/security-onion-solutions/so-fleet-launcher:2.3.70 (0)
	  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
									 Dload  Upload   Total   Spent    Left  Speed
	100   543  100   543    0     0   1447      0 --:--:-- --:--:-- --:--:--  1451
	The push refers to repository [igt-pgh-som-01:5000/security-onion-solutions/so-fleet-launcher]
	3913f643306c: Layer already exists
	dcd88013df4a: Layer already exists
	0e39e5a6c1a7: Layer already exists
	72fcdf866774: Layer already exists
	214bc1224e1f: Layer already exists
	28bfee8f2286: Layer already exists
	878dab86cf0f: Layer already exists
	2.3.70: digest: sha256:3b406ae8b1e393044b8f21e88a99d7e7d472384db5186c53d17cab11c75fc2ea size: 1795

- so-fleet is still missing from so-status
- I ran so-fleet-start and it started without error
- so-fleet is now showing is so-status, but it is still missing from my SOC.
- `docker container ls` shows the container using port 8080/tcp, but I'm unable to access the port using `wget --no-check-certificate https://localhost:8080` - I'm assuming that should be accessible and provide some result.

Any help with this will be very much appreciated.

Thank you, ...Rob

**Comment by dougburks** [comment, marked as answer]
When you ran Setup for this installation, are you sure that you enabled Fleet?

**Reply by Rbartram** [reply]
Duh!  [embarrassed]  I just re-ran so-setup and revisited menu options.   When moving to a distributed install, I neglected to consider the Fleet node.  

Can we just pretend this never happened?...

Thank you!

**Reply by Rbartram** [reply]
Hi. Doug, I just installed a Fleet node, however, Fleet is still missing from my SOC.  Running so-status on the Fleet node server shows that everything is ok.

Any pointers will be much appreciated.

Thank you, ...Rob
